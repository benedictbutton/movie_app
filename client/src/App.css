.App {
  background-color: #0b0b0b;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  font-size: calc(10px + 2vmin);
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  width: 100%;
  /* color: white; */
  border-bottom-style: solid;
  border-color: lightgrey;
}

.App-link {
  color: #61dafb;
}

.home {
  background-image: url("./assets/grey-large.jpg");
  background-repeat: no-repeat;
  -moz-background-size: 100% 100%;
  -webkit-background-size: 100% 100%;
  background-size: 100% 100%;
  flex: 1;
  width: 100%;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}

.theater {
  background-image: url("./assets/theater.jpg");
  background-repeat: no-repeat;
  -moz-background-size: 100% 100%;
  -webkit-background-size: 100% 100%;
  background-size: 100% 100%;
  display: flex;
  flex: 1;
  width: 100%;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  /* direction: row;
  position: relative; */
}

.sticky-container {
  width: 100%;
  position: relative;
}

.sticky-boundary {
  position: relative;
  margin: 0;
  min-height: 120vh;
}

/* .scrollbar::-webkit-scrollbar {
  overflow-y: scroll;
  height: 12px;
  width: 12px;
  background: yellow;
}

.scrollbar::-webkit-scrollbar-thumb {
  background: black;
  -webkit-border-radius: 1ex;
  -webkit-box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.75);
}

.scrollbar::-webkit-scrollbar-corner {
  background: #000;
} */

.sticky-box {
  position: relative;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1;
}

.sticky-box.is-top {
  top: 0px;
}

.sticky-box.is-top.is-sticky {
  box-shadow: 0 7px 2px -2px darkgray;
}

.sticky-box.is-bottom {
  bottom: 0px;
}
.sticky-box.is-bottom.is-sticky {
  box-shadow: 0 -7px 2px -2px darkgray;
}

.u-center-center {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.u-center-top {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
}

.u-feature-detect-sticky {
  position: absolute;
  position: sticky;

  position: -webkit-sticky;
  position: -moz-sticky;
  position: -ms-sticky;
  position: -o-sticky;
  position: sticky;
}

#image {
  background-image: url("./assets/grey-small.jpg");
  background-repeat: no-repeat;
  -moz-background-size: 100% 100%;
  -webkit-background-size: 100% 100%;
  background-size: 100% 100%;
}

#rotate {
  /* Safari */
  -webkit-transform: rotate(45deg);
  /* Firefox */
  -moz-transform: rotate(45deg);
  /* IE */
  -ms-transform: rotate(45deg);
  /* Opera */
  -o-transform: rotate(45sdeg);
  /* Internet Explorer */
  filter: progid:DXImageTransform.Microsoft.BasicImage(rotation=3);
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
