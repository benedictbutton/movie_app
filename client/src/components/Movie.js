import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Stars from "./Stars";
import Image from "../assets/brushed-metal.jpg";
//material-ui
import AddIcon from "@material-ui/icons/Add";
import Button from "@material-ui/core/Button";
import CheckIcon from "@material-ui/icons/Check";
import ClearIcon from "@material-ui/icons/Clear";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Tooltip from "@material-ui/core/Tooltip";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
  root: {
    flex: 1,
    justifyContent: "space-around",
    alignItems: "flex-start"
  },
  buttonOne: {
    color: "#E8E8E8",
    backgroundColor: "#E50000"
  },
  buttonTwo: {
    color: "#E8E8E8",
    backgroundColor: "#4B0082"
  },
  layout: {
    padding: theme.spacing.unit * 3,
    margin: theme.spacing.unit * 3,
    background: `url(${Image})`
  },
  grid: {
    height: "auto",
    padding: "56.25% 0 0 0",
    maxWidth: 300
  },
  media: {
    width: "100%",
    height: "100%"
  },
  stars: {
    background: "rgba(0, 0, 0, 0.5)",
    maxHeight: 48
  },
  overview: {
    color: "#343434"
  },
  title: {
    marginBottom: theme.spacing.unit * 3,
    color: "#2F4F4F",
    fontWeight: "700"
  }
});

const Movie = props => {
  const {
    classes,
    movie,
    imageUrl,
    check,
    handlePlaylistClick,
    handleRatingClick,
    activePlaylist,
    playlistMovieIds
  } = props;

  return (
    <Grid container display="flex">
      <Grid item xs={12} md={8}>
        <Paper className={classes.layout}>
          <Grid container className={classes.root} spacing={16}>
            <Grid item xs={12} sm={5} className={classes.grid}>
              <img src={imageUrl} className={classes.media} />
              <Paper className={classes.stars}>
                <Stars id={movie.id} movie={movie} starSize={20} />
              </Paper>
              <Grid container>
                <Grid item xs={6}>
                  <Tooltip title="Clear rating" placement="bottom">
                    <Button
                      className={classes.buttonOne}
                      component={Paper}
                      alt="clear rating"
                      fullWidth
                      onClick={() => {
                        handleRatingClick(movie);
                      }}
                    >
                      <ClearIcon />
                    </Button>
                  </Tooltip>
                </Grid>
                <Grid item xs={6}>
                  <Tooltip
                    title={check ? "Remove from playlist" : "Add to playlist"}
                    placement="bottom"
                  >
                    <Button
                      className={classes.buttonTwo}
                      component={Paper}
                      alt="clear rating"
                      fullWidth
                      onClick={() => {
                        handlePlaylistClick(check, activePlaylist, movie);
                      }}
                    >
                      {check ? <CheckIcon /> : <AddIcon />}
                    </Button>
                  </Tooltip>
                </Grid>
              </Grid>
            </Grid>
            <Grid item sm={7}>
              <Typography variant="h4" className={classes.title}>
                {movie.title || movie.name}
              </Typography>
              <Typography
                component="p"
                align="center"
                className={classes.overview}
              >
                {movie.overview || movie.description}
              </Typography>
            </Grid>
          </Grid>
        </Paper>
      </Grid>
    </Grid>
  );
};

export default withStyles(styles)(Movie);
