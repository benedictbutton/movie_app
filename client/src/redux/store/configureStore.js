import { createStore, applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import createSagaMiddleware from "redux-saga";
import rootReducer from "../reducers/index";
import rootSaga from "../sagas";
import { loadState, saveState } from "../../localStorage";
import throttle from "lodash/throttle";

const logger = createLogger();
const saga = createSagaMiddleware();
const persistedState = loadState();

const store = createStore(
  rootReducer,
  persistedState,
  applyMiddleware(saga, logger)
);

saga.run(rootSaga);

store.subscribe(
  throttle(() => {
    saveState({
      user: store.getState().user,
      ratings: store.getState().ratings
    });
  }, 1000)
);

export { store };
