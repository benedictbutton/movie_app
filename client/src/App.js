import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'
import './App.css'
//font awesome
import { library } from '@fortawesome/fontawesome-svg-core'
import {
    faSignOutAlt,
    faBars,
    faLock,
    faUser,
    faFilm,
    faStar,
    faStepForward,
    faPlay,
    faUpload,
} from '@fortawesome/free-solid-svg-icons'
//App Components
import Admin from './components/Admin'
import HeaderContainer from './containers/HeaderContainer'
import HomeContainer from './containers/HomeContainer'
import MovieContainer from './containers/MovieContainer'
import MoviesContainer from './containers/MoviesContainer'
import RatingsContainer from './containers/RatingsContainer'
import PlaylistContainer from './containers/PlaylistContainer'
import PlaylistsContainer from './containers/PlaylistsContainer'
import PrivateRoute from './HOC/PrivateRoute'
import ResetPasswordContainer from './containers/ResetPasswordContainer'
import SignInContainer from './containers/SignInContainer'
import SignUpContainer from './containers/SignUpContainer'
import UserContainer from './containers/UserContainer'
import MediaUpdate from './components/MediaUpdate'
import PersonContainer from './containers/PersonContainer'

import Fix from './components/Fix'

library.add(
    faSignOutAlt,
    faBars,
    faUser,
    faLock,
    faFilm,
    faStar,
    faStepForward,
    faPlay,
    faUpload
)

class App extends Component {
    render() {
        return (
            <div className="App">
                <Route exact path="/" component={HomeContainer} />
                <header className="App-header">
                    <Route path="/ms" component={HeaderContainer} />
                </header>
                <Switch>
                    <Route path="/ms/sign-up" component={SignUpContainer} />
                    <Route path="/ms/sign-in" component={SignInContainer} />
                    <Route
                        path="/ms/reset_password/:id/:pass"
                        component={ResetPasswordContainer}
                    />
                    <PrivateRoute path="/ms/admin" component={Admin} />
                    <PrivateRoute
                        path="/ms/movies/:type/:query?/:desc?/:end?"
                        component={MoviesContainer}
                    />
                    <PrivateRoute
                        path="/ms/movie/:id"
                        component={MovieContainer}
                    />
                    <PrivateRoute
                        path="/ms/ratings"
                        component={RatingsContainer}
                    />
                    <PrivateRoute
                        path="/ms/person/:id"
                        component={PersonContainer}
                    />
                    <PrivateRoute
                        path="/ms/playlists"
                        component={PlaylistsContainer}
                    />
                    <PrivateRoute
                        path="/ms/playlist/:id"
                        component={PlaylistContainer}
                    />
                    <PrivateRoute
                        path="/ms/users/:id"
                        component={UserContainer}
                    />
                    <Route path="/ms/media_update" component={MediaUpdate} />
                    <Route path="/ms/fix" component={Fix} />
                </Switch>
            </div>
        )
    }
}

export default App
//
