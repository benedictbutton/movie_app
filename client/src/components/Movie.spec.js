import React from "react";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import createSagaMiddleware from "redux-saga";
import { Provider } from "react-redux";
import { BrowserRouter as Router, MemoryRouter, Link } from "react-router-dom";
import App from "../App";
import AddIcon from "@material-ui/icons/Add";
import CheckIcon from "@material-ui/icons/Check";
import Movie from "./Movie";
import Typography from "@material-ui/core/Typography";
import {
  doPlaylistAddMovieRequesting,
  doPlaylistRemoveMovieRequesting
} from "../redux/actions/playlistActions";

const initialState = {
  client: {
    requesting: false,
    successful: false,
    admin: false,
    reset: false,
    notifications: { message: "test" }
  },
  user: {
    profile: {
      id: 7,
      firstName: "test",
      lastName: "user",
      username: "testuser",
      email: "testuser@email.com"
    },
    requesting: false,
    successful: false,
    notifications: {
      body: null,
      message: "test",
      code: 404,
      display: true
    }
  },
  movies: {
    results: [],
    list: {},
    cast: [],
    ratedList: {},
    searchList: {},
    requesting: false,
    successful: false,
    notifications: {},
    query: { type: "multi", page: 1, tag: "movie/popular" },
    filter: "all"
  },
  playlists: {
    active: 1,
    lists: [
      {
        active: true,
        id: 0,
        name: "Gremlins Documentary",
        user_id: 2
      }
    ],
    playlistMovieIds: { ["none"]: [], "1": [0] },
    playlistMovies: {},
    requesting: false,
    successful: false,
    notifications: {}
  }
};

const props = {
  movie: { id: 0, overview: "The never told before, true story of Gremlins" },
  profile: false,
  cast: [
    { id: 1, name: "Bill Burr" },
    { id: 2, name: "Emma Stone" },
    { id: 3, name: "Will Ferrell" }
  ],
  imageUrl: "https://image.tmdb.org/t/p/w500/BbNvKCuEF4SRzFXR16aK6ISFtR.jpg",
  check: true,
  activePlaylist: 1,
  handlePlaylistClick: jest.fn(() =>
    doPlaylistRemoveMovieRequesting(this.check, this.activePlaylist, this.movie)
  )
};

describe("Movie", () => {
  beforeEach(() => jest.resetAllMocks());

  test("renders consistently", () => {
    const wrapper = shallow(<Movie {...props} />);
    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();
    wrapper.unmount();
  });

  it("render()", () => {
    const wrapper = shallow(<Movie {...props} />).dive();
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.contains(<CheckIcon />)).toBe(true);
    expect(wrapper.contains(<AddIcon />)).toBe(false);
    wrapper.find("#btn").simulate("click");
    expect(props.handlePlaylistClick).toHaveBeenCalled();
    wrapper.setProps({ check: false });
    expect(wrapper.contains(<AddIcon />)).toBe(true);
    expect(
      wrapper
        .find(Typography)
        .at(3)
        .dive()
        .dive()
        .text()
    ).toBe("Bill Burr");

    wrapper.unmount();
  });
});
// expect(handlePlaylistClick()).toHaveBeenCalled(1);
