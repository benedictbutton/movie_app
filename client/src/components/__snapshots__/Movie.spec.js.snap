// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders consistently 1`] = `
<Fragment>
  <WithStyles(Grid)
    className="Movie-chip-4"
    container={true}
    item={true}
    xs={12}
  >
    <Connect(WithStyles(ActivePlaylistContainer)) />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    container={true}
    display="flex"
    justify="space-evenly"
  >
    <WithStyles(Grid)
      item={true}
      md={8}
      xs={12}
    >
      <WithStyles(Paper)
        className="Movie-layout-5"
      >
        <WithStyles(Grid)
          className="Movie-root-1"
          container={true}
          spacing={16}
        >
          <WithStyles(Grid)
            className="Movie-grid-6"
            item={true}
            sm={5}
            xs={12}
          >
            <img
              alt="movie poster"
              className="Movie-media-7"
              src="https://image.tmdb.org/t/p/w500/BbNvKCuEF4SRzFXR16aK6ISFtR.jpg"
            />
            <WithStyles(Paper)
              className="Movie-stars-8"
            >
              <Connect(WithStyles(Stars))
                id={0}
                movie={
                  Object {
                    "id": 0,
                    "overview": "The never told before, true story of Gremlins",
                  }
                }
                starSize={20}
              />
            </WithStyles(Paper)>
            <WithStyles(Grid)
              container={true}
            >
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Tooltip)
                  placement="bottom"
                  title="Clear rating"
                >
                  <WithStyles(Button)
                    alt="clear stars"
                    className="Movie-buttonOne-2"
                    component={[Function]}
                    fullWidth={true}
                    onClick={[Function]}
                  >
                    <pure(ClearIcon) />
                  </WithStyles(Button)>
                </WithStyles(Tooltip)>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Tooltip)
                  placement="bottom"
                  title="Remove from playlist"
                >
                  <WithStyles(Button)
                    alt="clear playlist"
                    className="Movie-buttonTwo-3"
                    component={[Function]}
                    fullWidth={true}
                    id="btn"
                    onClick={[Function]}
                  >
                    <pure(CheckIcon) />
                  </WithStyles(Button)>
                </WithStyles(Tooltip)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            sm={7}
          >
            <WithStyles(Typography)
              className="Movie-title-10"
              variant="h4"
            />
            <WithStyles(Typography)
              align="center"
              className="Movie-overview-9"
              component="p"
            >
              The never told before, true story of Gremlins
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </WithStyles(Paper)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="Movie-castLayout-13"
      item={true}
      md={3}
      xs={12}
    >
      <WithStyles(Typography)
        align="center"
        className="Movie-cast-11"
        variant="h6"
      >
        Cast
      </WithStyles(Typography)>
      <WithStyles(Grid)
        container={true}
      >
        <WithStyles(Grid)
          item={true}
          key="0"
          md={12}
          xs={4}
        >
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to={
              Object {
                "pathname": "/ms/person/1",
                "state": Object {
                  "imageUrl": "https://image.tmdb.org/t/p/w500/BbNvKCuEF4SRzFXR16aK6ISFtR.jpg",
                  "name": "Bill Burr",
                },
              }
            }
          >
            <WithStyles(Typography)
              className="Movie-members-12"
            >
              Bill Burr
            </WithStyles(Typography)>
          </Link>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          key="1"
          md={12}
          xs={4}
        >
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to={
              Object {
                "pathname": "/ms/person/2",
                "state": Object {
                  "imageUrl": "https://image.tmdb.org/t/p/w500/BbNvKCuEF4SRzFXR16aK6ISFtR.jpg",
                  "name": "Emma Stone",
                },
              }
            }
          >
            <WithStyles(Typography)
              className="Movie-members-12"
            >
              Emma Stone
            </WithStyles(Typography)>
          </Link>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          key="2"
          md={12}
          xs={4}
        >
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to={
              Object {
                "pathname": "/ms/person/3",
                "state": Object {
                  "imageUrl": "https://image.tmdb.org/t/p/w500/BbNvKCuEF4SRzFXR16aK6ISFtR.jpg",
                  "name": "Will Ferrell",
                },
              }
            }
          >
            <WithStyles(Typography)
              className="Movie-members-12"
            >
              Will Ferrell
            </WithStyles(Typography)>
          </Link>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
      <WithStyles(FullCast)
        cast={
          Array [
            Object {
              "id": 1,
              "name": "Bill Burr",
            },
            Object {
              "id": 2,
              "name": "Emma Stone",
            },
            Object {
              "id": 3,
              "name": "Will Ferrell",
            },
          ]
        }
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Fragment>
`;
